### PR TITLE
Side-effect free tests

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
The major change here is how we provision clean DB: mow it is done as stamping of SQLModel directly bypassing Alembic, which is faster (I can see about 20% decrease in tests runtime - not to be confused with overall CI runtime).
Also, I replaced `clean_workspace` with `clean_client` since workspaces will be out soon.

There is still some follow-up to be done with tests in that regard, but this will fall into subsequent PRs.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by strickvl — APPROVED** [review]
This looks good to me + really like the new way of initialising the DB when it's empty. I'd make sure to merge in the `mypy` update (from `develop`) before merging as that might raise some errors here.

**Review by fa9r — APPROVED** [review]
So far changes look all good to me 👍 

However please make sure that follow-up migrations still work with alembic if you initialize your DB from an intermediate state. If that's still working then this should be ready to go.

File: src/zenml/zen_stores/sql_zen_store.py [hunk 1]
```diff
@@ -991,8 +991,24 @@ def migrate_database(self) -> None:
...
+                with Session(self.engine) as session:
+                    session.add(
+                        IdentitySchema(
+                            id=str(GlobalConfiguration().user_id).replace(
+                                "-", ""
+                            )
+                        )
+                    )
+                    session.commit()
+                self.alembic.stamp("head")
```
[the remark this answers is not in the document]
Comment (avishniakov) [reply src/zenml/zen_stores/sql_zen_store.py:1011]: Yeah, it stamps the Alembic version as the current migration head id. I will do an extra verification of this to be double-sure.

[the remark this answers is not in the document]
Comment (avishniakov) [reply src/zenml/zen_stores/sql_zen_store.py:1011]: Yep, upgrade works as expected

**Pull request merged**